### PR TITLE
Added dictionary.js

### PR DESCRIPTION
**Pull request by KlasafGeijerstam**
A key generator that uses a dictionary in CSV format to create its keys.
Example dictionary:

Hastebin,
Is,
Awesome,
Kappa

### CODE REVIEW

File: lib/key_generators/dictionary.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  //Options
+  if (!options) 
+    return done(Error('No options passed to generator'));
+  if(!options.path)
+    return done(Error('No dictionary path specified in options'));
+  
+  //Load dictionary
+  fs.readFile(options.path,'utf8',(err,data) => {
+    if(err) throw err;
+    dictionary = data.split(',');
```
Comment (seejohnrun) [inline lib/key_generators/dictionary.js]: I think this should actually be `split('\n')` and the file should be one word per line

File: lib/key_generators/dictionary.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+var DictionaryGenerator = function(options) {
+  //Options
+  if (!options) 
+    return done(Error('No options passed to generator'));
+  if(!options.path)
+    return done(Error('No dictionary path specified in options'));
+  
+  //Load dictionary
+  fs.readFile(options.path,'utf8',(err,data) => {
```
Comment (seejohnrun) [inline lib/key_generators/dictionary.js]: spaces after argument `,` please

File: lib/key_generators/dictionary.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    if(err) throw err;
+    dictionary = data.split(',');
+    
+    //Remove any non alpha-numeric characters
+    for(var i = 0; i < dictionary.length; i++){
```
Comment (seejohnrun) [inline lib/key_generators/dictionary.js]: space after `for` and before `{`, although I think you should be able to remove this line given the above

File: lib/key_generators/dictionary.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      dictionary[i] = dictionary[i].replace(/\W/g,'');
+    }
+
+    random = rand.integer(0,dictionary.length);
+    randomEngine = rand.engines.nativeMath;
```
Comment (seejohnrun) [inline lib/key_generators/dictionary.js]: Should be able to set this in line with the instantiation above

File: lib/key_generators/dictionary.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  fs.readFile(options.path,'utf8',(err,data) => {
+    if(err) throw err;
+    dictionary = data.split(',');
+    
+    //Remove any non alpha-numeric characters
+    for(var i = 0; i < dictionary.length; i++){
+      dictionary[i] = dictionary[i].replace(/\W/g,'');
+    }
+
+    random = rand.integer(0,dictionary.length);
```
Comment (seejohnrun) [inline lib/key_generators/dictionary.js]: this should be a property on `this` given that it it directly tied to this dictionary

**Comment by seejohnrun** [comment]
Also you'll want to add `random-js` to the package if you're going to use it

File: lib/key_generators/dictionary.js (hunk 1 above)
Comment (KlasafGeijerstam) [inline lib/key_generators/dictionary.js]: If you split on '\n, wouldn't that mean that files using windows-style line endings will have an extra \r at the end of every entry.? Is it better to split on '\n' and then removing the potential extra non alpha-numerics, or should one assume that the dictionary contains just '\n' and no other non alpha-numerics?

Comment (seejohnrun) [inline lib/key_generators/dictionary.js]: You can also just split on /[\n\r]+/ :)

Comment (KlasafGeijerstam) [inline lib/key_generators/dictionary.js]: Smart! I changed it to your regex, and now it works with both kinds of line endings

File: lib/key_generators/dictionary.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+var fs = require('fs')
+var dictionary;
```
Comment (seejohnrun) [inline lib/key_generators/dictionary.js]: I believe this line isn't being used

**Comment by KlasafGeijerstam** [comment]
Did some minor changes, no longer dependent on random-js, and splits on newlines instead of , .

File: lib/key_generators/dictionary.js (hunk 6 above)
Comment (KlasafGeijerstam) [inline lib/key_generators/dictionary.js]: Correct, removed it!

**Comment by seejohnrun** [comment]
Thanks!

**Pull request merged**